SU2_CFD/include/solver_structure.hpp

```diff
@@ -3595,6 +3596,51 @@ class CSolver {
                         unsigned short val_kind_solver,
                         unsigned short val_kind_marker);
 
+  /*!
+   * \brief Load a spanwise inlet profile data from file into a particular solver.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver - Container vector with all of the solvers.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] val_iter - Current external iteration number.
+   * \param[in] val_kind_solver - Solver container position.
+   * \param[in] val_kind_marker - Kind of marker to apply the profiles.
+   */
+  void LoadSpanwiseInletProfile(CGeometry **geometry,
+                                CSolver ***solver,
+                                CConfig *config,
+                                int val_iter,
+                                unsigned short val_kind_solver,
+                                unsigned short val_kind_marker);
+
+  /*!
+   * \brief Get Pi (as (f(x+1)-f(x))/delta_x) for Akima interpolation for inlet file
+   * \param[in] iRow - Current matching row in inlet file for interpolation to Vertex
+   * \param[in] index - Current interpolating column in inlet file
+   */
+  inline passivedouble Get_Pi(unsigned long iRow, unsigned short index);
+
+    /*!
+   * \brief Get the weights Wi for Akima interpolation for inlet file 
+   * \param[in] iRow - Current matching row in inlet file for interpolation to Vertex
+   * \param[in] index - Current interpolating column in inlet file
+   */
+  inline passivedouble Get_Wi(unsigned long iRow,unsigned short index);
+
+    /*!
+   * \brief Get the weighed slope of the interval being interpolated by Akima.
+   * \param[in] iRow - Current matching row in inlet file for interpolation to Vertex
+   * \param[in] index - Current interpolating column in inlet file
+   * \param[in] jMarker - Current Marker to retrive the number of Rows for that Marker
+   */
+  inline passivedouble Get_Ai_dash(unsigned long iRow, unsigned short index,unsigned long jMarker);
```

Comment:
These would probably go to an interpolation class (toolbox) instead.

Reply:
So you want me to define all these functions in the interpolation class (interpolation_structure.cpp)? Won't it be simpler to stay in the solver class given I am calling everything from there?

Reply:
No not in interpolation_structure, that is for interfacing between zones.
Somewhere in Common/toolboxes as a generic 1D interpolation framework.

For now those functions are only used in solver but we should try to make things as modular as possible.

Reply:
Right. That would mean that I would have to make and #include a new .hpp file in solver_structure.hpp?

Reply:
That's right.